### ISSUE TITLE
Timber method get_widgets - FATAL ERROR

### ISSUE THREAD

**Issue by dawidryba** [timber/timber#1412]
I have problem with new Timber with Twig 2.0. When i try get widgets with code:
```
$this->context['widget_footer'] = Timber::get_widgets('widget-footer');
$this->context['widget_phone'] = Timber::get_widgets('widget-phone');
```

i got error:
```
 Fatal error: Uncaught LogicException: Function "dynamic_sidebar" is already registered. in /home/dawidryba/projekty/praca/sowdent.dev/wp-content/themes/krframe/krFrame/vendor/twig/twig/lib/Twig/Extension/Staging.php:30 Stack trace: #0 /home/dawidryba/projekty/praca/sowdent.dev/wp-content/themes/krframe/krFrame/vendor/twig/twig/lib/Twig/ExtensionSet.php(162): Twig_Extension_Staging->addFunction(Object(Twig_SimpleFunction)) #1 /home/dawidryba/projekty/praca/sowdent.dev/wp-content/themes/krframe/krFrame/vendor/twig/twig/lib/Twig/Environment.php(807): Twig_ExtensionSet->addFunction(Object(Twig_SimpleFunction)) #2 /home/dawidryba/projekty/praca/sowdent.dev/wp-content/themes/krframe/krFrame/vendor/timber/timber/lib/FunctionWrapper.php(70): Twig_Environment->addFunction(Object(Twig_SimpleFunction)) #3 /home/dawidryba/projekty/praca/sowdent.dev/wp-includes/class-wp-hook.php(298): Timber\FunctionWrapper->add_to_twig(Object(Twig_Environment)) #4 /home/dawidryba/projekty/praca/sowdent.dev/wp-includes/plugin.php(203): WP_Hook->apply_ in /home/dawidryba/projekty/praca/sowdent.dev/wp-content/themes/krframe/krFrame/vendor/twig/twig/lib/Twig/Extension/Staging.php on line 30
```

If i comment one get_widgets then work fine.

I check method get_widgets before update and difference is:

OLD:
```
public static function get_widgets( $widget_id ) {
      return trim(Helper::function_wrapper('dynamic_sidebar', array($widget_id), true));
}
```
NEW:
```
public static function get_widgets( $widget_id ) {
		$output = new FunctionWrapper( 'dynamic_sidebar', array( $widget_id ), true );
		return trim( $output );
}
```

**Comment by danimbrogno** [comment]
I've got this problem as well. Not knowing much about this my impression is that  the dynamic_sidebar function is getting created each time the get_widgets call is run and it's getting created in some shared scope.

My work around is to catch the original dynamic_sidebar function with an output buffer:

```
public function getSidebar($id) {

    ob_start();

    dynamic_sidebar( $id );

    $html = ob_get_contents();

    ob_end_clean();

    return $html;

  }

  public function context($context) {

    $context['posts'] = Timber::get_posts();
    $context['pagination'] = Timber::get_pagination();
    
    $context['headerWidgets'] = $this->getSidebar(WidgetArea\HomeHeaderWidgetArea::$id);
    $context['bodyWidgets'] = $this->getSidebar(WidgetArea\HomeBodyWidgetArea::$id);

    return $context;

  }
```

**Comment by simonmilz** [comment]
I can confirm this error too.

**Comment by gchtr** [comment]
Until this is fixed, you can try to use the following:

```
$this->context['widget_footer'] = Timber\Helper::ob_function( 'dynamic_sidebar', array( 'widget-footer' ) );
$this->context['widget_phone'] = Timber\Helper::ob_function( 'dynamic_sidebar', array( 'widget-phone' ) );
```

What @danimbrogno came up with as a solution is actually already built into Timber. [`Helper::ob_function()`](http://timber.github.io/timber/#ob_function) is thought for functions that normally echo output. It catches the output and returns it, so that you can add it to context.

**Comment by jarednova** [comment]
Closed by #1422
